### ISSUE TITLE
Feature: Bulk Section creation #17 - duplicate sections were created

### ISSUE THREAD

**Issue by melindakraft** [tl-its-umich-edu/canvas-course-manager-next#196]
I tested with two CSV files on this course: https://canvas-test.it.umich.edu/courses/211644/settings
When attempting to create with duplicate section names, the validation partially worked (it didn't report all as duplicate). 

When I tried again with a new CSV file it created all the sections even though there were all duplicates. I think that the tool is hanging on to some session info that is inaccurate or incomplete when trying to run it a second time.

Attaching the 2 files I was testing with. 

[sections2.csv](https://github.com/tl-its-umich-edu/canvas-course-manager-next/files/7205631/sections2.csv)
[sections.csv](https://github.com/tl-its-umich-edu/canvas-course-manager-next/files/7205632/sections.csv)

**Comment by pushyamig** [comment]
@ssciolla I am assigning this to you since you noticed this your self with the PR you are working on. Try to use the course and sample csv files Melinda we using.

**Comment by ssciolla** [comment]
@pushyamig, if you could circle back to ensure the language still makes sense in Create Sections and that duplicates work correctly, that would be helpful.

**Comment by pushyamig** [comment]
I did some research on this issue. I feel the problem in duplicate section creation is due to file encoding.

The file shared in this issue `section.csv` and `section2.csv` are normal csv file. There is one common section name as `aarohee` in both these file but one with space (in `section.csv`) and not in other file (`section2.csv). 

Some how it is excel or Mac default editor adding a coded `?` symbol with trailing space. So when user uploads that file the content of that file is already `aarohee?` (with weird encoded `?`). I was thinking Browser Files API or `papaparse` may be intervening before the react code. But that was not the case

When I saved the same file `section.csv` as UTF-8 encoded option, the training space was still there when you look at the file in Excel but when I uploaded it CCM for creating section it did not had encoded `?` weird symbol. So this seen as `aarohee `(with trailing space). We have code in place that is trimming the leading/training spaces. 
```
file.text().then(
      (v) => {
        const results = Papa.parse<CSVRecord>(v.trim(), this.parseConfig)
        onComplete(results.meta.fields, results.data)
      }```

Here is the sample `section.csv` with UTF encoded.
[sections.csv](https://github.com/tl-its-umich-edu/canvas-course-manager-next/files/8050081/sections.csv)

I did another test, I created 2 csv files not from excel but using text editor both having same section name one with space another with out space and saved it as csv file. When I uploaded the files to CCM the section name is not tampered.
[spaces.csv](https://github.com/tl-its-umich-edu/canvas-course-manager-next/files/8050095/spaces.csv)
[spaces2.csv](https://github.com/tl-its-umich-edu/canvas-course-manager-next/files/8050096/spaces2.csv)

So conclusion I feel is the issue is happening as excel encoded the space if saved without UTF-8 encoding. I think this is edge case. I am not sure if we want to allow users only UTF-8 encoded csv files.

**Comment by melindakraft** [comment]
I don't think we don't want to try and work around the strange nuances of Excel encoding. It seems safer just to trim off leading and trailing spaces. I agree that we don't want to limit submission to UTF-8 encoding because that will cause more problems for end users who don't understand encoding options.

**Comment by pushyamig** [comment]
Thanks @melindakraft I think I am going to close this issue as won't fix.
